index.js

```diff
@@ -344,6 +345,45 @@ app.post('/api/doorTest', async(req, res) => {
     await handleSensorRequest(locationid)
 });
 
+app.post('/api/door', async(request, response) => {
+    const coreId = request.body.coreid
+    const locationid = db.getLocationFromCoreId(coreId)
+  
+    const message = JSON.parse(request.body.data)
+    const signal = message.data
+    var doorSignal
+  
+    if (signal==IM21_DOOR_STATUS.OPEN){
```

Comment:
For `open` and `closed`, you should probably use `DOOR_STATE` from `SessionStateDoorEnum.js`.

For `LowBatt` and `HeartBeat`, it doesn't look like these are being used for anything. You're just sending `doorSignal` to `addIM21DoorSensorData` which only cares if it's `open` or `closed`. I suggest that you just remove lines 362-367. In these cases, it will just send `undefined` as `doorSignal` to `addIM21DoorSensorData` which will then do nothing. Would that do what you want here?

Reply:
So my rationale for doing it this way was thus - 

- The door:locationID key on redis is the logical place to add data from the sensor such as low battery/heartbeat - which are definitely data points that we want to log. 

- However, the current statemachine would need to be modified before we can have entries other than `open` or `closed` in this stream. 

- as I was implementing this, I had to choose between ignoring any signal data other than `00` or `02` **or** parse it and then choose to not add it to the stream. Given the above two points, I chose the latter as it signalled an intention to support using low battery/heartbeat data when we overhaul the state machine. 

I do see your point about it being a stub that leads to nowhere, and I suspect that is frowned upon in code? If so, I'll delete these lines...

Reply:
No, I think that you have a good point. If it'll help us remember to do it later and it's accurate, then it seems OK to leave it there. Thanks for explaining.